web/src/filter.ts

```diff
@@ -579,7 +579,13 @@ export class Filter {
    These are not keys in a JavaScript object, because we
    might need to support multiple terms of the same type.
 */
-    static unparse(search_terms: NarrowTerm[]): string {
+    static unparse(
+        search_terms: NarrowTerm[],
+        // If the `unparse` is being done to just generate
+        // operator suggestions, we don't need to parse
+        // operands. See `search_suggestion.ts` for related logic.
```

Comment:
Just a note -- I'm not entirely sure this will prove accurate always, in that an existing `is:dm` operator will have a different impact on what other operators are appropriate than an `is:starred` operator, but I'm not sure this function is called in a context where that matters.